src/common/libflux/stats.h

```diff
@@ -0,0 +1,79 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _FLUX_CORE_STATS_H
+#define _FLUX_CORE_STATS_H
+
+#include "handle.h"
+
+/* Metric types:
+ * Counter - An integer value that will, on the backend (brubeck) send the
+ *           count and reset to 0 at each flush. It calculates the change
+ *           from the value sent at the previous flush.
+ *           An example of where to use a counter is the builtin msgcounters
+ *           that are part of each flux_t handle. The counts will continually
+ *           increase and brubeck will handle calculating the count sent in
+ *           the interval.
+ * Gauge   - An integer that takes an arbitrary value and maintains its
+ *           it is set again. Gauges can also take incremental values
+ *           in the form of a + or - in front of the value which increments
+ *           the previously stored value.
+ *           An example of where to use a gauge is to track the current
+ *           size of the broker's content-cache. At each point, the cache's
+ *           sizes are independent of each other.
+ * Timer   - A double value which represents the time taken for a given
+ *           task in ms.
+ *           An example of where to use a timer is timing the length of
+ *           asynchronous loads in the broker's content-cache. The cache
+ *           entry can keep track of when the load was started and then
+ *           calculate and send the time taken once the entry is loaded.
+ */
+
+/* Update (or create) and store 'count' for 'name' to be sent on the
+ * next flush.
+ */
+void flux_stats_count (flux_t *h, const char *name, ssize_t count);
+
+/* Update (or create) and store 'value' for 'name' to be sent on the
+ * next flush.
+ */
+void flux_stats_gauge_set (flux_t *h, const char *name, ssize_t value);
+
+/* Update (or create) and increment 'value' for 'name' to be sent on the
+ * next flush. If'name' was not previously stored, then the value is stored
+ * directly (i.e. assumed 0 previous value).
+ */
+void flux_stats_gauge_inc (flux_t *h, const char *name, ssize_t inc);
+
+
+/* Update (or create) and store 'ms' for 'name' to be sent on the
+ * next flush.
+ */
+void flux_stats_timing (flux_t *h, const char *name, double ts);
```

Comment:
Comment references `ms` but argument is `ts`.  (Its not a timestamp, so ts might be misleading)

Name of function is actually `flux_stats_timer()` right?

Reply:
Function is named `flux_stats_timing` (and `fripp_timing`) I believe to make it clear that fripp/stats don't keep track of the timer internally.

Reply:
My bad, I thought I read "timer" in the C code and "timing" in the header.  You're right.